src/handlers/next/create-app-register-handler.ts

```diff
@@ -22,6 +56,49 @@ export type CreateAppRegisterHandlerOptions = HasAPL & {
    * or a function that receives a full Saleor API URL ad returns true/false.
    */
   allowedSaleorUrls?: Array<string | ((saleorApiUrl: string) => boolean)>;
+  /**
+   * Run right after Saleor calls this endpoint
+   */
+  onRequestStart?(
+    request: Request,
+    context: {
+      authToken?: string;
+      saleorDomain?: string;
+      saleorApiUrl?: string;
+      respondWithError: typeof createCallbackError;
+    }
+  ): Promise<void>;
+  /**
+   * Run after all security checks
+   */
+  onRequestVerified?(
+    request: Request,
+    context: {
+      authData: AuthData;
+      respondWithError: typeof createCallbackError;
+    }
+  ): Promise<void>;
+  /**
+   * Run after APL successfully AuthData, assuming that APL.set will reject a Promise in case of error
+   */
+  onAuthAplSaved?(
+    request: Request,
+    context: {
+      authData: AuthData;
+      respondWithError: typeof createCallbackError;
+    }
+  ): Promise<void>;
+  /**
+   * Run after APL fails to set AuthData
+   */
+  onAuthAplFailed?(
+    request: Request,
+    context: {
+      authData: AuthData;
```

Comment:
If the APL has failed, are the authData available?

Reply:
>If the APL has failed, are the authData available?

Yes, this is argument for `set` so you can still receive it